libs/model/src/main/java/de/cyface/model/RawRecord.java

```diff
@@ -142,4 +142,13 @@ public void setSpeed(double speed) {
     public void setModality(Modality modality) {
         this.modality = modality;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        if (!super.equals(o)) return false;
+        RawRecord rawRecord = (RawRecord) o;
+        return Double.compare(rawRecord.accuracy, accuracy) == 0 && Double.compare(rawRecord.speed, speed) == 0 && modality == rawRecord.modality;
+    }
```

Comment:
I could also have converted the class to Kotlin but didn't really touch the class - I just noticed there is no equals method which could have been the cause for a bug I noticed (the cause was another).